ecc/components/product-selector/product-selector.js

```diff
@@ -56,16 +55,36 @@ export default class ProductSelector extends LitElement {
     return this.selectedProduct.name && !isEmptyObject(this.selectedProduct);
   }
 
+  getAvailableProducts() {
+    const selectedProducts = this.existingProducts || [];
+    return this.products.filter((product) => {
+      const notSelectedByOthers = !selectedProducts.some((p) => p.name === product.name);
+      const isCurrentProduct = product.name === this.selectedProduct.name;
+      return notSelectedByOthers || isCurrentProduct;
+    })
+      .map((product) => html`<sp-menu-item value="${product.name}">${product.title}</sp-menu-item>`);
+  }
+
+  getImageSource() {
+    const { tagImage, name } = this.selectedProduct;
+    const match = this.products.find((product) => product.name === name);
+    return tagImage || match?.tagImage || '/ecc/icons/icon-placeholder.svg';
+  }
+
   render() {
-    const match = this.products.find((product) => product.name === this.selectedProduct.name);
+    const { name, title, showProductBlade } = this.selectedProduct;
+    const availableProducts = this.getAvailableProducts();
+    const imageSource = this.getImageSource();
+
     return html`
       <fieldset class="rsvp-field-wrapper">
-      ${html`<img class="product-img" src="${this.selectedProduct.tagImage || match?.tagImage || '/ecc/icons/icon-placeholder.svg'}" alt="${this.selectedProduct.title || nothing}">`}  
-        <sp-picker class="product-select-input" label="Select a product" value=${this.selectedProduct.name || nothing} @change="${this.handleSelectChange}">
-          ${this.products.map((product) => html`<sp-menu-item value="${product.name}">${product.title}
-          </sp-menu-item>`)}
+        <img class="product-img" src="${imageSource}" alt="${title || nothing}">
+        <sp-picker class="product-select-input" label="Select a product" value=${name || nothing} @change="${this.handleSelectChange}">
+          ${availableProducts}
         </sp-picker>
-        ${html`<sp-checkbox class="checkbox-product-link" .checked=${this.selectedProduct.showProductBlade} .disabled=${!this.isValidSelection()} @change="${this.handleCheckChange}">Show ${this.selectedProduct.title || '[Product name]'} product blade on event details page</sp-checkbox>`}
+        <sp-checkbox class="checkbox-product-link" .checked=${showProductBlade} .disabled=${!this.isValidSelection()} @change="${this.handleCheckChange}">
+          Show ${title || '[Product name]'} product blade on event details page
```

Comment:
should this come from placeholder since we will have localization in phase 2 and different locale context ecc path would have to show different message for the marketer in that language?

Reply:
I was under the impression that ECC interface will forever be in English. We will have a large list of things to placeholder-ize if that wasn't true

Reply:
yeah we would have /locale/ecc/create/t3 and /locales/dashbaord/t3 to have the fetch based on that locale to display the translated events from en_us or to have ability to create that language specific event which will not get cross translated.